### PR TITLE
Fix closure warnings with MIN_WEBGL_VERSION=2. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #22373

### CODE REVIEW

**Comment by floooh** [comment]
Giving this a whirl, one sec...

File: test/test_other.py [hunk 1]
```diff
@@ -9479,6 +9480,7 @@ def test_closure_full_js_library(self, args):
       '--minify=0',
       '-Werror=closure',
       '-sINCLUDE_FULL_LIBRARY',
+      '-sOFFSCREEN_FRAMEBUFFER',
```
Comment (floooh) [inline test/test_other.py:9483]: Is this change intended?

**Comment by floooh** [comment]
I can confirm that the PR fixes the Closure warnings I'm seeing, and spot checking my sokol-samples (https://floooh.github.io/sokol-html5/) I'm also not seeing any runtime problems.

Only question is about the `-sOFFSCREEN_FRAMEBUFFER` change, I'm just wondering if this is an unintended change (see my comment above).

**Review by floooh — APPROVED** [review]

**Comment by sbc100** [comment]
> I can confirm that the PR fixes the Closure warnings I'm seeing, and spot checking my sokol-samples (https://floooh.github.io/sokol-html5/) I'm also not seeing any runtime problems.
> 
> Only question is about the `-sOFFSCREEN_FRAMEBUFFER` change, I'm just wondering if this is an unintended change (see my comment above).

That is just an artifact of what the test is test for here.   The test is checking for an non-minified `.deleteVertexArray` which (when GL=2) is only part of the output in `LEGACY_GL_EMULATION` mode.  Basically this setting is just increasing the amount of code that is generated to include more things.

**Comment by floooh** [comment]
Thanks for the quick fix btw @sbc100 :)

File: src/library_webgl.js [hunk 2]
```diff
@@ -1864,11 +1867,13 @@ for (/**@suppress{duplicate}*/var i = 0; i <= {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
       return;
     }
 #endif
+#if INCLUDE_WEBGL1_FALLBACK
```
Comment (kripken) [inline src/library_webgl.js:1870]: It looks like this is reached even without the condition on `settings.MIN_WEBGL_VERSION`, unlike all those before it, so this if is not actually for the complement condition, IIUC?

Comment (sbc100) [reply src/library_webgl.js:1870]: WEBGL_USE_GARBAGE_FREE_APIS itself is conditional on MIN_WEBGL_VERSION.

The idea is that INCLUDE_WEBGL1_FALLBACK will be set only in cases where `WEBGL_USE_GARBAGE_FREE_APIS` is set and also we know that we have only webgl2 support.  In this case that means that above return on line 1867 is guaranteed to run.

Comment (sbc100) [reply src/library_webgl.js:1870]: The exact same pattern repeats below on line 1886

**Comment by sbc100** [comment]
Looks like we have some code size savings as a result of this too

**Comment by sbc100** [comment]
> Looks like we have some code size savings as a result of this too

Sorry .. not true.

File: tools/link.py [hunk 3]
```diff
@@ -1659,6 +1659,8 @@ def phase_linker_setup(options, state, newargs):
     # For chrome see: https://crbug.com/324992397
     if settings.MIN_CHROME_VERSION != feature_matrix.UNSUPPORTED and settings.MEMORY64 and settings.MAXIMUM_MEMORY > 2 ** 32:
       settings.WEBGL_USE_GARBAGE_FREE_APIS = 0
+    if settings.WEBGL_USE_GARBAGE_FREE_APIS and settings.MIN_WEBGL_VERSION >= 2:
```
Comment (kripken) [inline tools/link.py:1662]: Can this check be simplified, given that `WEBGL_USE_GARBAGE_FREE_APIS` is only set when `MIN_WEBGL_VERSION >= 2` anyhow?

Comment (kripken) [reply tools/link.py:1662]:
```suggestion
    if settings.WEBGL_USE_GARBAGE_FREE_APIS:
      assert(settings.MIN_WEBGL_VERSION >= 2)
```

File: src/library_webgl.js (hunk 2 above)
Comment (kripken) [reply src/library_webgl.js:1870]: I see, thanks, I was confused by the condition in link.py. I wrote a suggestion to simplify it below.

**Review by kripken — COMMENTED** [review]
lgtm % suggestion

File: tools/link.py (hunk 3 above)
Comment (sbc100) [reply tools/link.py:1662]: Are you confusing `MAX_WEBGL_VERSION` (which is the setting which `WEBGL_USE_GARBAGE_FREE_APIS` depends on) with `MIN_WEBGL_VERSION` which is used here?

Comment (kripken) [reply tools/link.py:1662]: Ah, yes, that was exactly my confusion, thanks!
